### ISSUE TITLE
"VM has reported a failure when processing extension 'vm0-dfinstallGW'

### ISSUE THREAD

**Issue by srcsakthivel** [Azure/azure-quickstart-templates#8071]
[101-vms-with-private-selfHostIR](https://github.com/Azure/azure-quickstart-templates/tree/master/101-vms-with-selfhost-integration-runtime)

### Issue Details
{
    "status": "Failed",
    "error": {
        "code": "ResourceDeploymentFailure",
        "message": "The resource operation completed with terminal provisioning state 'Failed'.",
        "details": [
            {
                "code": "VMExtensionProvisioningError",
                "message": "VM has reported a failure when processing extension 'vm0-dfinstallGW'. Error message: \"Command execution finished, but failed because it returned a non-zero exit code of: '1'. The command had an error output of: ' CategoryInfo          : OperationStopped: (Failed to run p...1.MoveNext( \r\n   )\r\n.:String) [], RuntimeException\r\n    + FullyQualifiedErrorId : Failed to run process: exitCode=-2146233088, err \r\n   Variable=, errContent=, outContent=Failed to execute the c...' For more information, check the instance view by executing Get-AzVmssVm or Get-AzVm (https://aka.ms/GetAzVm). These commands can be executed using CloudShell (https://aka.ms/CloudShell)\"\r\n\r\nMore information on troubleshooting is available at https://aka.ms/VMExtensionCSEWindowsTroubleshoot "
            }
        ]
    }
}

### Repro steps (*if necessary, delete otherwise*)
1. Pre-provisioned Data Factory instance, VNET and Subnet
2. [101-vms-with-private-selfHostIR](https://github.com/Azure/azure-quickstart-templates/tree/master/101-vms-with-selfhost-integration-runtime) >> [Deploy to azure](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F101-vms-with-selfhost-integration-runtime%2Fazuredeploy.json) - by fill in the information.

**Comment by srcsakthivel** [comment]
Digging further I got the following information from the trace log,

Microsoft.DataTransfer.DIAgentClient.HostServiceException: Failed to get service token from ADF service with key IR@********-****-****-****-************@for-shir@ServiceEndpoint=for-shir.eastus2.datafactory.azure.net@****************** and time cost is: 3.4063888 seconds, the error code is: UnexpectedFault, activityId is: 1768e8d1-cb39-4ea0-95f4-c9b1735f43e1 and detailed error message is An error occurred while sending the request.
The remote name could not be resolved: 'for-shir.eastus2.datafactory.azure.net'

**Comment by MCKLMT** [comment]
Can you try again? There was a transient error from DataFactory in this region. Thanks

**Comment by srcsakthivel** [comment]
Thanks @MCKLMT, it worked in Central US. 

Due to the Enterprise connectivity setup, we have to stick with **East US 2**. May I know how to get the issue addressed?

**Comment by MCKLMT** [comment]
I suggest to try again in East US 2 and if it fails to open a support request.

**Comment by srcsakthivel** [comment]
ok, Thanks!

**Comment by MCKLMT** [comment]
@srcsakthivel Can you share news?

**Comment by dgumpper** [comment]
This error is now happening in both East US and Central US. The deployment template keeps generating this error every time it is used. Is there a manual method to deploy the GW? Thanks

**Comment by MCKLMT** [comment]
I suggest to open a support request for troubleshooting.

**Comment by esuomi** [comment]
For what it's worth, I'm just today seeing this on **West Europe** with the same conditions as in original post. This template worked fine yesterday. Is there any additional information available?

**Comment by srcsakthivel** [comment]
No, it worked on retry with no change. As @MCKLMT mentioned you can open a Microsoft ticket.
